File: plugins/dotnet/robocode.dotnet.robotscs/src/tested/robotscs/MaxVelocity.cs
/**
 * Copyright (c) 2001-2019 Mathew A. Nelson and Robocode contributors
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * https://robocode.sourceforge.io/license/epl-v10.html
 */
using Robocode;

namespace tested.robotscs
{
    public class MaxVelocity : AdvancedRobot
    {
        public override void Run()
        {
            for (;;)
            {
                int time = (int) Time;

                if (time < 200)
                {
                    Out.WriteLine(time + ": " + Velocity);
                }

                switch (time)
                {
                    case 1:
                        SetAhead(1000);
                        break;

                    case 20:
                        SetBack(1000);
                        break;

                    case 40:
                        MaxVelocity = (4.2);
                        SetAhead(1000);
                        break;

                    case 60:
                        SetBack(1000);
                        break;

                    case 80:
                        MaxVelocity = (100);
                        SetAhead(1000);
                        break;

                    case 100:
                        SetBack(1000);
                        break;

                    case 120:
                        MaxVelocity = (-2);
                        SetAhead(1000);
                        break;

                    case 140:
                        SetBack(1000);
                        break;

                    case 160:
                        MaxVelocity = (0);
                        SetAhead(1000);
                        break;

                    case 180:
                        SetBack(1000);
                        break;
                }
                Execute();
            }
        }
    }
}

File: robocode.tests.robots/src/main/java/tested/robots/MaxVelocity.java
/**
 * Copyright (c) 2001-2019 Mathew A. Nelson and Robocode contributors
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * https://robocode.sourceforge.io/license/epl-v10.html
 */
package tested.robots;


/**
 * @author Flemming N. Larsen (original)
 */
public class MaxVelocity extends robocode.AdvancedRobot {

	public void run() {
		for (;;) {
			int time = (int) getTime();

			if (time < 200) {
				out.println(time + ": " + getVelocity());
			}

			switch (time) {
			case 1:
				setAhead(1000);
				break;

			case 20:
				setBack(1000);
				break;

			case 40:
				setMaxVelocity(4.2);
				setAhead(1000);
				break;

			case 60:
				setBack(1000);
				break;

			case 80:
				setMaxVelocity(100);
				setAhead(1000);
				break;				

			case 100:
				setBack(1000);
				break;

			case 120:
				setMaxVelocity(-2);
				setAhead(1000);
				break;

			case 140:
				setBack(1000);
				break;

			case 160:
				setMaxVelocity(0);
				setAhead(1000);
				break;

			case 180:
				setBack(1000);
				break;
			}
			execute();
		}
	}
}
